### DISCUSSION TITLE
the configure item "SCORE_THRESH_TEST " doesn't work for a configure file generated by lazyconfig_train_net.py

### DISCUSSION THREAD

**Discussion by proshanm** [facebookresearch/detectron2#5083] [Q&A]
I finetuned a MAE pretrained model for object detection using ViTDet project using our own dataset by involing lazyconfig_train_net.py. I wanted to load the finetuned module and visualize the result, but I found a problem that the structure of configure file 'config.yaml' generated by lazyconfig_train_net.py is different from that of configure file of build-in module. So I implemented user-defined Predictor which load module and weight from config.yaml instead of DefaultPredictor and visualize the output using Visualizer. The result could be visualized, but there were many low-score boxes that I didn't want to show. so I add hierarchical property MODEL.ROI_HEADS.SCORE_THRESH_TEST into config.yaml, but it doesn't work. I know the problem can be solved by modifying Visualizer to introduce a score thresh, but i'm not sure if it is necessory for this case or there's another better way for solving the problem. The code of predictor is as follow.
`class ProjectPredictor:

    def __init__(self, cfg):
        self.model = instantiate(cfg.model)
        self.model.eval()
        if len(cfg.dataloader.test):
            self.metadata = MetadataCatalog.get(cfg.dataloader.test.dataset['names'])

        checkpointer = DetectionCheckpointer(self.model)
        checkpointer.load(cfg.train.init_checkpoint)

        self.aug = T.ResizeShortestEdge(
            [cfg.dataloader.test.mapper.augmentations[0]['short_edge_length'], cfg.dataloader.test.mapper.augmentations[0]['short_edge_length']], cfg.dataloader.test.mapper.augmentations[0]['max_size']
        )

        self.input_format = cfg.dataloader.test.mapper.image_format
        assert self.input_format in ["RGB", "BGR", 'L'], self.input_format

    def __call__(self, original_image):
        """
        Args:
            original_image (np.ndarray): an image of shape (H, W, C) (in BGR order).

        Returns:
            predictions (dict):
                the output of the model for one image only.
                See :doc:`/tutorials/models` for details about the format.
        """
        with torch.no_grad():  # https://github.com/sphinx-doc/sphinx/issues/4258
            # Apply pre-processing to image.
            if self.input_format == "RGB":
                # whether the model expects BGR inputs or RGB
                original_image = original_image[:, :, ::-1]
            height, width = original_image.shape[:2]
            image = self.aug.get_transform(original_image).apply_image(original_image)
            # 
            if self.input_format == "L":
                image = image.astype("float32")
                image = [image]
                image = torch.as_tensor(image)
            else:
                image = torch.as_tensor(image.astype("float32").transpose(2, 0, 1))
            inputs = {"image": image, "height": height, "width": width}
            predictions = self.model([inputs])[0]
            return predictions`

**Comment by bmaxdk** [comment]
You can filtering out low-scoring bounding boxes with modify the predictor. 
You can directly modify the prediction outputs in your custom predictor after calling `self.model([inputs])[0]`. The returned predictions dict usually contains a key **instances**, which itself contains various fields like `scores`. You can filter out boxes with low scores before returning them.

```py
threshold = 0.5  # set your own threshold
instances = predictions["instances"]
keep = instances.scores > threshold  # assuming the field is named 'scores'
instances = instances[keep]
predictions["instances"] = instances

```
